packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -30,18 +31,35 @@ const clientVersionHeaderKey = 'apollographql-client-version';
 export const plugin = <TContext>(
   options: EngineReportingOptions<TContext> = Object.create(null),
   addTrace: (args: AddTraceArgs) => Promise<void>,
-  // schemaHash: string,
+  startSchemaReporting: ({
+    executableSchema,
+    executableSchemaId,
+  }: {
+    executableSchema: string;
+    executableSchemaId: string;
+  }) => void,
+  executableSchemaIdGenerator: (schema: string | GraphQLSchema) => string,
 ): ApolloServerPlugin<TContext> => {
   const logger: Logger = options.logger || console;
   const generateClientInfo: GenerateClientInfo<TContext> =
     options.generateClientInfo || defaultGenerateClientInfo;
 
-
   return {
+    serverWillStart: function({ schema }) {
+      if (options.experimental_schemaReporting) {
+        startSchemaReporting({
+          executableSchema:
+            options.experimental_overrideReportedSchema ||
+            printSchema(schema),
+          executableSchemaId:
+            executableSchemaIdGenerator(options.experimental_overrideReportedSchema || schema)
+        });
+      }
+    },
```

Comment:
Since we don't do anything else if `experimental_schemaReporting` is disabled, might we just do the following so we could de-dent most of this block?:

```suggestion
    serverWillStart: function({ schema }) {
      if (!options.experimental_schemaReporting) return;

      startSchemaReporting({
        executableSchema:
          options.experimental_overrideReportedSchema ||
          printSchema(schema),
        executableSchemaId:
          executableSchemaIdGenerator(options.experimental_overrideReportedSchema || schema)
      });
    },
```

(And loose some braces.)